packages/auth/src/accessToken/routes.ts

```diff
@@ -91,18 +91,19 @@ async function revokeToken(
   deps: ServiceDependencies,
   ctx: RevokeContext
 ): Promise<void> {
+  const token = (ctx.headers['authorization'] ?? '').replace('GNAP ', '')
   const { id: managementId } = ctx.params
-  await deps.accessTokenService.revoke(managementId)
+  await deps.accessTokenService.revoke(managementId, token)
```

Comment:
Would it be possible to check access tokens in a separate middleware (maybe the signature middleware)?
The GNAP spec talks about binding client keys to access tokens (both **access token** and continue access tokens). Could `client` and `clientKeyId` ([on its way out](https://github.com/interledger/rafiki/issues/737)) and maybe even continue details be moved off of grants, which would avoid doing grant service lookups in signature middleware
https://github.com/interledger/rafiki/blob/83a6a546f9d4f2e830db261973e4cffeeaedd828/packages/auth/src/signature/middleware.ts#L151
https://github.com/interledger/rafiki/blob/83a6a546f9d4f2e830db261973e4cffeeaedd828/packages/auth/src/signature/middleware.ts#L84-L89

Reply:
Moving `client` & `clientKeyId` into a new table would make sense since they may be bound to multiple grants anyways. In the case of continue details it can be done but I think it's a little trickier to justify because it'd punt the call to a different, more narrow service instead since they're one-to-one with grants.

What service calls are "in-scope" for the signature (or some other) middleware? As I understand it it seems like the idea is to just lookup via the client service if the token/grant continuation is associated with the client via its bound key.

Reply:
Maybe there can be path specific auth middlewares (ahead of the signature middleware) that store `client` on `ctx` like the `backend` currently does (via the grant)
https://github.com/interledger/rafiki/blob/ec15723671d951d6ea0657176b4c8d664be97eca/packages/backend/src/open_payments/auth/middleware.ts#L63
Some of these could store other route specific data like the access token or the grant on the `ctx`. The backend does this with payment pointer routes
https://github.com/interledger/rafiki/blob/ec15723671d951d6ea0657176b4c8d664be97eca/packages/backend/src/open_payments/payment_pointer/middleware.ts#L20
https://github.com/interledger/rafiki/blob/ec15723671d951d6ea0657176b4c8d664be97eca/packages/backend/src/open_payments/payment_pointer/routes.ts#L26-L37
Then there can be a single signature middleware.

Reply:
Going to capture this for now in https://github.com/interledger/rafiki/issues/882, in case I don't get to it before the holidays.